src/main/java/org/cbioportal/service/impl/ClinicalEventServiceImpl.java

```diff
@@ -108,4 +122,101 @@ public List<ClinicalEventTypeCount> getClinicalEventTypeCounts(List<String> stud
+                    .map(endPositionIdentifier::applyAsInt)
+                    .map(endDate -> startPositionIdentifier.applyAsInt(event) < endDate)
+                    .orElse(true)
+            ).toList();
+
+        filteredStudyIds.clear();
+        filteredPatientIds.clear();
+        // get all the studies that have start events but not end events
+        for (ClinicalEvent clinicalEvent : patientStartEvents.stream().filter(event -> !patientEndEventsById.containsKey(getKey(event))).toList()) {
+            filteredStudyIds.add(clinicalEvent.getStudyId());
+            filteredPatientIds.add(clinicalEvent.getPatientId());
+        }
+
+        List<ClinicalEvent> patientCensoredEvents = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(filteredStudyIds)) {
+            patientCensoredEvents = clinicalEventRepository.getTimelineEvents(filteredStudyIds, filteredPatientIds, censoredClinicalEventsMeta);
+        }
+
+        Map<String, ClinicalEvent> patientCensoredEventsById = patientCensoredEvents.stream().collect(Collectors.toMap(ClinicalEventServiceImpl::getKey, Function.identity()));
+
+        return patientStartEvents.stream()
+            .flatMap(event -> {
+                ClinicalData clinicalDataMonths = new ClinicalData();
+                clinicalDataMonths.setStudyId(event.getStudyId());
+                clinicalDataMonths.setPatientId(event.getPatientId());
+                clinicalDataMonths.setAttrId(attributeIdPrefix + "_MONTHS");
+
+                ClinicalData clinicalDataStatus = new ClinicalData();
+                clinicalDataStatus.setStudyId(event.getStudyId());
+                clinicalDataStatus.setPatientId(event.getPatientId());
+                clinicalDataStatus.setAttrId(attributeIdPrefix + "_STATUS");
+
+                int startDate = startPositionIdentifier.applyAsInt(event);
+                int endDate;
+                if (patientEndEventsById.containsKey(getKey(event))) {
+                    clinicalDataStatus.setAttrValue("1:EVENT");
+                    endDate = endPositionIdentifier.applyAsInt(patientEndEventsById.get(getKey(event)));
+                } else {
+                    // ignore cases where patient does not have censored timeline events or
+                    // stop date of start event is less than start date of censored events
+                    if (!patientCensoredEventsById.containsKey(getKey(event)) ||
+                        startDate >= censoredPositionIdentifier.applyAsInt(patientCensoredEventsById.get(getKey(event)))
+                    ) {
+                        return Stream.empty();
+                    }
+
+                    clinicalDataStatus.setAttrValue("0:CENSORED");
+                    endDate = censoredPositionIdentifier.applyAsInt(patientCensoredEventsById.get(getKey(event)));
+                }
+                clinicalDataMonths.setAttrValue(String.valueOf((endDate - startDate) / 30.4));
```

Comment:
I would set this to some const, same for the other "magic" strings

Reply:
got it